### PR TITLE
[SNAP-1721] Avoid code generation failure

### PR DESCRIPTION
**Pull request by rishitesh**
Code generation for SpecificSafeProjection fails for the above-mentioned example if one uses dataSet.collect() instead of dataSet.show()

## Changes proposed in this pull request
The failure is due to type mismatch between immutable.map and scala.collection.Map.
Changed the example to use scala.collection.Map interface.

## Patch testing

Manual testing

## ReleaseNotes.txt changes

NA

## Other PRs 

NA

### CODE REVIEW

File: examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala [hunk 1]
```diff
@@ -18,73 +18,100 @@ package org.apache.spark.examples.snappydata
...
+  * bin/snappy-job.sh submit
+  * --app-name WorkingWithObjects
+  * --class org.apache.spark.examples.snappydata.WorkingWithObjects
+  * --app-jar examples/jars/quickstart.jar
+  * --lead [leadHost:port]
+  *
+  * Check the status of your job id
+  * bin/snappy-job.sh status --lead [leadHost:port] --job-id [job-id]
+  *
+  */
```
Comment (kneeraj) [inline examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala]: Formatting got changed accidentally. Revert it.

File: examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala [hunk 2]
```diff
@@ -18,73 +18,100 @@ package org.apache.spark.examples.snappydata
...
+    // Create a columnar table with the a Struct to store Address
     snSession.sql("CREATE table Persons(name String, address Struct<city: String, state:String>, " +
         "emergencyContacts Map<String,String>) using column options()")
 
     // Write the created DataFrame to the columnar table.
     people.write.insertInto("Persons")
 
-    //print schema of the table
+    // print schema of the table
+    // scalastyle:off println
```
Comment (kneeraj) [inline examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala]: remove these scalastype:off commented out lines

File: examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala [hunk 3]
```diff
@@ -18,73 +18,100 @@ package org.apache.spark.examples.snappydata
...
     println("Print Schema of the table\n################")
     println(snSession.table("Persons").schema)
     println
-
+    // scalastyle:on println
```
Comment (kneeraj) [inline examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala]: similar to above.

File: examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala (hunk 2 above)
Comment (rishitesh) [inline examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala]: This is to suppress the scale style error if somebody puts a println line.

File: examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala (hunk 1 above)
Comment (rishitesh) [inline examples/src/main/scala/org/apache/spark/examples/snappydata/WorkingWithObjects.scala]: This is intentional to keep in sync with the formatting of other classes.

**Pull request merged**